src/screens/DashboardScreen.tsx

```diff
@@ -44,28 +48,48 @@ const DashboardScreen = ({ navigation, style }: ScreenProps) => {
   const activeWalletName = useAppSelector((s) => s.activeWallet.name)
   const networkStatus = useAppSelector((s) => s.network.status)
   const networkName = useAppSelector((s) => s.network.name)
+  const { scrollDirection } = useScrollContext()
 
   const refreshData = () => {
     if (!isLoading) dispatch(syncAddressesData(addressHashes))
   }
 
+  const animatedButtonStyle = useAnimatedStyle(() => ({
+    width: withTiming(scrollDirection?.value === 'down' ? 56 : 170)
+  }))
+
+  const animatedTextStyle = useAnimatedStyle(() => ({
+    opacity: withTiming(scrollDirection?.value === 'down' ? 0 : 1)
+  }))
+
   return (
-    <DashboardScreenStyled refreshControl={<RefreshControl refreshing={isLoading} onRefresh={refreshData} />}>
-      <ScreenSectionStyled>
-        <SurfaceHeader>
-          <AppText color="primary" semiBold size={30}>
-            {activeWalletName}
-          </AppText>
-          <ActiveNetwork>
-            <NetworkStatusBullet status={networkStatus} />
-            <AppText color="primary">{networkName}</AppText>
-          </ActiveNetwork>
-        </SurfaceHeader>
-
-        <BalanceSummaryStyled dateLabel="VALUE TODAY" />
-      </ScreenSectionStyled>
-      <AddressesTokensListStyled />
-    </DashboardScreenStyled>
+    <>
+      <DashboardScreenStyled refreshControl={<RefreshControl refreshing={isLoading} onRefresh={refreshData} />}>
+        <ScreenSectionStyled>
+          <SurfaceHeader>
+            <AppText color="primary" semiBold size={30}>
+              {activeWalletName}
+            </AppText>
+            <ActiveNetwork>
+              <NetworkStatusBullet status={networkStatus} />
+              <AppText color="primary">{networkName}</AppText>
+            </ActiveNetwork>
+          </SurfaceHeader>
+
+          <BalanceSummaryStyled dateLabel="VALUE TODAY" />
+        </ScreenSectionStyled>
+        <AddressesTokensListStyled />
+      </DashboardScreenStyled>
+      <AnimatedFloatingButton style={animatedButtonStyle}>
+        <FloatingButton Icon={ArrowUpDown} round color="white" onPress={() => navigation.navigate('ReceiveNavigation')}>
```

Comment:
I hooked it up to the dashboard floating icon for now, in a next PR, once we decide on the mockups, I'll implement the more advanced floating button.

Reply:
Sounds good!